jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/LambdaLineMarker.kt

```diff
@@ -64,6 +72,11 @@ class LambdaLineMarker : LineMarkerProviderDescriptor() {
     ) {
     }
 
+    private fun isHandlerValid(element: PsiElement, handler: String, runtime: Runtime): Boolean =
+        (runtime.runtimeGroup == RuntimeGroup.JAVA && !handler.contains("::")) || // Java class level handler is valid
```

Comment:
We should push this language specific piece down to the Java extension point; which would give other runtimes a hook too which will enable them to override when the line marker should be shown.

Go for example would probably always want to show the line markers because the requirements for them are extremely specific to Lambda - and so it's obviously a Lambda.

Reply:
Good point, I've added another method in the `LambdaHandlerResolver` to handle language level check.